src/Ext/TechnoType/Body.cpp

```diff
@@ -32,6 +32,33 @@ void TechnoTypeExt::ExtData::ApplyTurretOffset(Matrix3D* mtx, double factor)
 	mtx->Translate(x, y, z);
 }
 
+// Checks if a transport can load a passenger.
+// Note that this function only checks the size limit and Ares passenger whitelist and blacklist,
+// it doesn't check if this transport is actually a transport or not.
+bool TechnoTypeExt::ExtData::CanLoadPassenger(TechnoClass* pTransport, TechnoClass* pPassenger) const
+{
+	auto const pTransportType = pTransport->GetTechnoType();
+	auto const pPassengerType = pPassenger->GetTechnoType();
+	return (pTransportType->SizeLimit <= 0 || pTransportType->SizeLimit >= pPassengerType->Size)
+		&& (!this->Passengers_BySize || (pTransport->Passengers.GetTotalSize() + pPassengerType->Size) <= pTransportType->Passengers)
+		&& (this->PassengersWhitelist.empty() || this->PassengersWhitelist.Contains(pPassengerType))
+		&& !this->PassengersBlacklist.Contains(pPassengerType);
+}
+
+bool TechnoTypeExt::ExtData::CanLoadAny(TechnoClass* pTransport, std::vector<TechnoClass*> pPassengerList) const
+{
+	for (auto pPassenger : pPassengerList)
+	{
+		if (this->CanLoadPassenger(pTransport, pPassenger))
+		{
+			return true;
+		}
+	}
```

Comment:
Think this one can be allowed an exception like this:

```suggestion
	for (auto pPassenger : pPassengerList)
		if (this->CanLoadPassenger(pTransport, pPassenger))
			return true;

```